demux/packet.h

```diff
@@ -83,4 +89,19 @@ int demux_packet_add_blockadditional(struct demux_packet *dp, uint64_t id,
 
 void demux_packet_unref_contents(struct demux_packet *dp);
 
+struct demux_packet_pool {
+    mp_mutex lock;
+    struct demux_packet *packets;
+    struct MPContext *parent;
+};
```

Comment:
I would make this definition private

Reply:
I need a definition to keep instance in mpv_global

Reply:
it could be a pointer but doesn't really matter.

Reply:
Then we have to add `demux_packet_pool_alloc`. Seems little bit excessive for internal use or struct with two fields.